sklearn/utils/estimator_checks.py

```diff
@@ -156,6 +156,30 @@ def check_supervised_y_no_nan(name, estimator_orig):
                          "array y with NaN value.".format(name))
 
 
+def check_constant_features(name, estimator_orig):
+    # Checks that estimators work with constant features
+    # or raise a reasonable error message
+
+    # create features with a constant column
+    rng = np.random.RandomState(888)
+    X = np.abs(rng.randn(10, 5))
+    X[:, 1] = 1
+    X = pairwise_estimator_convert_X(X, estimator_orig)
+    # create target
+    y = np.arange(10) % 2
+    estimator = clone(estimator_orig)
+    y = enforce_estimator_tags_y(estimator, y)
+
+    try:
+        estimator.fit(X, y)
+    except FloatingPointError as e:
+        errmsg = 'The system is too ill-conditioned for this solver'
+        if errmsg not in str(e):
+            raise ValueError("Estimator {0} raised error as expected, but "
```

Comment:
Shall I add a test for this exception with `pytest.raises`?

Reply:
How can I make `codecov` not fail on this line?

Reply:
When checking that estimators failing with constant features return the correct error message, I changed the `if`-`raise` for an `assert`. After that, the coverage test was successful.